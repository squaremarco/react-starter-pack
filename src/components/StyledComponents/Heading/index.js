import React from 'react';
import styled from 'styled-components';

const StyledHeading = styled.h1`
  text-align: center;
  color: #4c4c4c;
`;

const Heading = (props) => {
  const { children } = props;
  return <StyledHeading>{children}</StyledHeading>;
};

export default Heading;
