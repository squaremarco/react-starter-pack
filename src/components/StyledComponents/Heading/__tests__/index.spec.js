import React from 'react';
import { render, cleanup } from '@testing-library/react';

import Heading from '..';

afterEach(cleanup);

describe('Heading', () => {
  const message = 'test';
  const component = <Heading>{message}</Heading>;

  it('Renders text correctly', () => {
    const { findByText } = render(component);
    expect(findByText(message)).not.toBeNull();
  });

  it('Should apply default styles', () => {
    const { container } = render(component);
    expect(container.firstChild).toHaveStyleRule('color', '#4c4c4c');
  });

  it('Snapshot test', () => {
    const { asFragment } = render(component);
    expect(asFragment()).toMatchSnapshot();
  });
});
