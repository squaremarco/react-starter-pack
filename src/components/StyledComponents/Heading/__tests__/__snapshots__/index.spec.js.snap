// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading Snapshot test 1`] = `
.c0 {
  text-align: center;
  color: #4c4c4c;
}

<div>
  <h1
    class="c0"
  >
    test
  </h1>
</div>
`;
